### DISCUSSION TITLE
I want to use NextPageContext for all pages

### DISCUSSION THREAD

**Discussion by ryo-ryukalice** [vercel/next.js#12919] [Help] [answered]
Both English and Next.js are beginners. sorry.

I want to refer to cookies on all pages.
The library for using cookies wants `NextPageContext`.
In `getInitialProps` in _app.tsx, the type of context is `AppContext` instead of `NextPageContext`.

```tsx
MyApp.getInitialProps = async (ctx: AppContext) => {
  const token = parseCookies(ctx)['api_token']
  // => Error: Argument of type 'AppContextType<Router>' is not assignable to parameter of type 'NextPageContext'.

  store.dispatch(authSlice.actions.login({ token: token }))

  const appProps = await App.getInitialProps(ctx)
  return { ...appProps }
}
```

Is there a way to access Cookies using `NextPageContext` on all pages?

**Comment by lfades** [comment, marked as answer]
Try changing it to `parseCookies(ctx.ctx)['api_token']`?

**Reply by ryo-ryukalice** [reply]
great! I got a token!
Thank you so much!
